### PR TITLE
audiowaveform: init at 1.4.2

### PR DESCRIPTION
**Pull request by edbentley**
###### Motivation for this change

Adding the [audiowaveform](https://github.com/bbc/audiowaveform) CLI program. Used their [homebrew file](https://github.com/bbc/homebrew-audiowaveform/blob/master/audiowaveform.rb) as a reference.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/audio/audiowaveform/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, cmake, boost, gd, libsndfile, libmad, libid3tag }:
+
+stdenv.mkDerivation rec {
+  pname = "audiowaveform";
+  version = "1.4.2";
+
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/tools/audio/audiowaveform/default.nix:7]: Can you please use fetchGitHub?

This makes cherry-picking fixes easier.

File: pkgs/tools/audio/audiowaveform/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://github.com/bbc/audiowaveform/archive/${version}.tar.gz";
+    sha256 = "0m13hhngz3p2nm9z5rq6pkm5xy0d7qg6hvm76pvmn3mfs6avkiiz";
+  };
+
+  buildInputs = [cmake boost gd libsndfile libmad libid3tag];
```
Comment (SuperSandro2000) [inline pkgs/tools/audio/audiowaveform/default.nix:12]:
```suggestion
  buildInputs = [ cmake boost gd libsndfile libmad libid3tag ];
```

**Comment by edbentley** [comment]
@SuperSandro2000 thanks for the quick feedback, updated.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101947` [1](https://github.com/Mic92/nixpkgs-review) run on x86_64-darwin
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>audiowaveform</li>
  </ul>
</details>

File: pkgs/tools/audio/audiowaveform/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = "bbc";
+    repo = "audiowaveform";
+    rev = version;
+    sha256 = "0k2s2f2hgq4pnjzfkgvjwgsflihmzdq7shicfjn0z2mzw4d1bvp2";
+  };
+
+  buildInputs = [ cmake boost gd libsndfile libmad libid3tag ];
```
Comment (sikmir) [inline pkgs/tools/audio/audiowaveform/default.nix:14]: Move `cmake` to `nativeBuildInputs`.

File: pkgs/tools/audio/audiowaveform/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with stdenv.lib; {
+    description = "C++ program to generate waveform data and render waveform images from audio files";
+    longDescription = ''
+      audiowaveform is a C++ command-line application that generates waveform data from either MP3, WAV, FLAC, or Ogg Vorbis format audio files.
+      Waveform data can be used to produce a visual rendering of the audio, similar in appearance to audio editing applications.
+    '';
+    homepage = "https://github.com/bbc/audiowaveform";
+    changelog = "https://github.com/bbc/audiowaveform/blob/${version}/ChangeLog";
+    license = licenses.gpl3;
```
Comment (sikmir) [inline pkgs/tools/audio/audiowaveform/default.nix:26]:
```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/tools/audio/audiowaveform/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "bbc";
+    repo = "audiowaveform";
+    rev = version;
+    sha256 = "0k2s2f2hgq4pnjzfkgvjwgsflihmzdq7shicfjn0z2mzw4d1bvp2";
+  };
+
+  buildInputs = [ cmake boost gd libsndfile libmad libid3tag ];
+
+  cmakeFlags = [ "-DENABLE_TESTS=0" ];
```
Comment (sikmir) [inline pkgs/tools/audio/audiowaveform/default.nix:16]: Is it possible to enable tests?

Comment (edbentley) [reply pkgs/tools/audio/audiowaveform/default.nix:16]: Building the tests requires [Google Test](https://github.com/google/googletest) which I didn't see a package for so I left it out (could always be worked on in a follow-up PR). Their [homebrew file](https://github.com/bbc/homebrew-audiowaveform/blob/master/audiowaveform.rb) also leaves out tests.

**Comment by edbentley** [comment]
@sikmir Updated, thanks for the feedback.

File: pkgs/tools/audio/audiowaveform/default.nix (hunk 5 above)
Comment (sikmir) [reply pkgs/tools/audio/audiowaveform/default.nix:18]: `gtest` package: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/gtest/default.nix

Comment (edbentley) [reply pkgs/tools/audio/audiowaveform/default.nix]: Oh nice, added 👍

**Comment by edbentley** [comment]
I think everything is ready here, can this be merged? 😄

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -2407,6 +2407,12 @@
     githubId = 984691;
     name = "Evan Danaher";
   };
+  edbentley = {
+    email = "hello@edbentley.dev";
+    github = "edbentley";
+    githubId = 15923595;
+    name = "Ed Bentley";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2414]: Please put this into a separate commit with the message: ``maintainers: add edbentley``

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
and please squash the current 4 commits together. After that we can merge.

**Comment by SuperSandro2000** [comment]
Also please squash the review related commits into the init ones.

**Comment by SuperSandro2000** [comment]
One test is failing:

```
[       OK ] SndFileAudioFileReaderTest.shouldNotProcessFileMoreThanOnce (1 ms)
[ RUN      ] SndFileAudioFileReaderTest.shouldReportErrorIfNotAWavFile
/build/source/test/SndFileAudioFileReaderTest.cpp:316: Failure
Value of: error.str()
Expected: is equal to "Failed to read file: ../test/data/test_file_stereo.mp3\nFile contains data in an unknown format.\n"
  Actual: "Failed to read file: ../test/data/test_file_stereo.mp3\nFormat not recognised.\n"
[  FAILED  ] SndFileAudioFileReaderTest.shouldReportErrorIfNotAWavFile (0 ms)
[----------] 10 tests from SndFileAudioFileReaderTest (5 ms total)
```

**Comment by edbentley** [comment]
@SuperSandro2000 How are you running the tests? When I run `nix-build -A audiowaveform` it shows as passed to me?

```
Running tests...
/nix/store/z7qvx6wbvmjvs767g06h03pkzr5aqn6i-cmake-3.18.2/bin/ctest --force-new-ctest-process
Test project /private/var/folders/bh/4nbf2dn14wn3ztnrbrqn11l80000gn/T/nix-build-audiowaveform-1.4.2.drv-0/source/build
    Start 1: audiowaveform_tests
1/1 Test #1: audiowaveform_tests ..............   Passed    6.03 sec

100% tests passed, 0 tests failed out of 1
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>audiowaveform</li>
  </ul>
</details>

Failed test

**Comment by SuperSandro2000** [comment]
I run nixpkgs-review pr 101947.

**Comment by edbentley** [comment]
Thanks, reproduced locally now. So the failing test is the error message should say `"File contains data in an unknown format"` but instead says `"Format not recognised"`...

I'm not sure where that error originates from but I don't see it mentioned in the codebase, so perhaps a dependency. I'm not an original contributor to [audiowaveform](https://github.com/bbc/audiowaveform) so I wouldn't know where to start on correcting this test. If the tests can be disabled I can progress, otherwise I think this is as far as I can reasonably get here on my own.

**Comment by SuperSandro2000** [comment]
> If the tests can be disabled I can progress, otherwise I think this is as far as I can reasonably get here on my own.

Yeah, you can just disable the tests or the entire package on darwin like ``broken ? stdenv.isDarwin;``.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101947` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>audiowaveform</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>audiowaveform</li>
  </ul>
</details>

```
/build/source/test/SndFileAudioFileReaderTest.cpp:316: Failure
Value of: error.str()
Expected: is equal to "Failed to read file: ../test/data/test_file_stereo.mp3\nFile contains data in an unknown format.\n"
  Actual: "Failed to read file: ../test/data/test_file_stereo.mp3\nFormat not recognised.\n"
  [  FAILED  ] SndFileAudioFileReaderTest.shouldReportErrorIfNotAWavFile (0 ms)
```

**Comment by edbentley** [comment]
Tests have been disabled (added a comment on why) and I've squashed the commits.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101947` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>audiowaveform</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>audiowaveform</li>
  </ul>
</details>

**Pull request merged**